File: server/src/main/java/com/fluffy/reaction/domain/LikeQueryService.java
package com.fluffy.reaction.domain;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class LikeQueryService {

    private final ReactionRepository reactionRepository;

    @Transactional(readOnly = true)
    public boolean isLiked(Like like, Long memberId) {
        if (memberId == null || memberId == -1) {
            return false;
        }

        return reactionRepository.existsByTargetTypeAndTargetIdAndMemberId(
                like.target().name(),
                like.targetId(),
                memberId
        );
    }
}

File: server/src/main/kotlin/com/fluffy/reaction/domain/LikeQueryService.kt
package com.fluffy.reaction.domain

import com.fluffy.global.web.Accessor
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class LikeQueryService(
    private val reactionRepository: ReactionRepository,
) {

    @Transactional(readOnly = true)
    fun isLiked(like: Like, accessor: Accessor): Boolean {
        if (accessor.isGuest()) return false

        return reactionRepository.findByTargetTypeAndTargetIdAndMemberIdAndType(
            like.target.name,
            like.targetId,
            accessor.id,
            ReactionType.LIKE,
        )?.status == ReactionStatus.ACTIVE
    }
}
